sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,538 @@
+until all invocations are completed.
+
+To provide more information to the compiler and to relax the requirement for
+`get` to be invoked implicitly, the property `balanced` may be specified on a
+`task_sequence` object, which guarantees that a user will not allow a destructor
+on that `task_sequence` object to be called when there are outstanding `async`
+invocations that have not been balanced by a matching `get` call. On spatial
+architectures, in the presence of this property, potentially expensive hardware
+may be elided. It is undefined behavior to specify the `balanced` property on
+`task_sequence` and then to allow the `task_sequence` object to be destroyed
+while there are any `async` invocations for which `get` has not been called.
+
+== `task_sequence` Properties
+
+The following code and table describe the properties that can be provided when
+declaring a `task_sequence` object.
+
+[source,c++,linenums]
+----
+namespace sycl::ext::intel::experimental {
+struct balanced_key {
+  using value_t = property_value<balanced_key>;
+};
+
+struct invocation_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<invocation_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+struct response_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<response_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+inline constexpr balanced_key::value_t balanced;
+template <uint32_t Size>
+inline constexpr invocation_capacity_key::value_t<Size> invocation_capacity;
+template <uint32_t Size>
+inline constexpr response_capacity_key::value_t<Size> response_capacity;
+
+template <> struct is_property_key<balanced_key> : std::true_type {};
+template <> struct is_property_key<invocation_capacity_key> : std::true_type {};
+template <> struct is_property_key<response_capacity_key> : std::true_type {};
+
+template <auto &f, typename PropertyListT>
+struct is_property_key_of<balanced_key, task_sequence<f, PropertyListT>>
+  : std::true_type {};
+template <auto &f, typename PropertyListT>
+struct is_property_key_of<invocation_capacity_key, task_sequence<f, PropertyListT>>
+  : std::true_type {};
+template <auto &f, typename PropertyListT>
+struct is_property_key_of<response_capacity_key, task_sequence<f, propertiesT>>
+  : std::true_type {};
+
+} // namespace sycl::ext::intel::experimental
+----
+
+--
+[options="header"]
+|===
+| Property | Description
+| balanced | The `balanced` property is a guarantee to the SYCL device compiler
+that the `task_sequence` object will call exactly the same number of `async` s
+and `get` s over the object's lifetime (i.e. before the `task_sequence`
+destructor is invoked).
+
+| invocation_capacity | The `async` invocations are guaranteed to not block the
+caller as long as the number of outstanding invocations are less than or equal
+to `invocation_capacity`. An outstanding invocation is an inflight
+`task_sequence` invocation that has not yet completed.
```

Comment:
Shouldn't this be:
```suggestion
`task_sequence` invocation that has not yet had get() called on it.
```
As worded it sounds like it only requires f() to have completed execution, meaning that you must physically use an infinitely large FIFO to store asynced calls, which would defeat the point on the invocation_capacity property.